app/domain/calculate_orders.py

```diff
@@ -0,0 +1,52 @@
+from datetime import datetime, timedelta
+
+import pytz
+from typing import List, Optional, Dict, Type, Generic, TypeVar, Callable
+
+T = TypeVar('T')
+
+
+class CalculateOrders:
+    """
+    This class responsible for calculating orders based on an existing forecast
+    """
+
+    @classmethod
+    async def process(
+            cls,
+            forecast: Optional[Dict[str, dict]],
+            recipe_model: Generic[T],
```

Comment:
This called `injection`: when we inject an object in runtime to function or method in order to use object in runtime.

It's probably OK for this specific case (I would go with constructor-based injection though). But let's try at least define types for models.

It's not OK that domain logic uses word "model" from the `db` layer.

We should rather:

1. Define type for repository - object with access to DB
2. Make `process` method depend on the type, not actual implementation details
3. Inject instance of repository class (that may use models inside) on the layer above